src/core/lexer/tokens.js

```diff
@@ -0,0 +1,112 @@
+  Indent:               null,
+  Dedent:               null,
+  EndOfLine:            /\n/,
+  EndOfInput:           null,
+
+  // Literals in ("<<commands>>")
+  Number:               /-?[0-9]+(\.[0-9+])?/,
+  String:               /"([^"\\]*(?:\\.[^"\\]*)*)"/,
+
+  // Command syntax ("<<foo>>")
+  BeginCommand:         /<</,
+  EndCommand:           />>/,
+
+  // Variables ("$foo")
+  Variable:             /\$([A-Za-z0-9_.])+/,
+
+  // Shortcut syntax ("->")
+  ShortcutOption:       /->/,
+
+  // Hashtag ("#something")
+  Hashtag:              /#([^(\s|#|//)]+)/, // seems a little hacky to explicitly consider comments here
+
+  // Comment ("// some stuff")
+  Comment:              /\/\/.*/,
+
+  // Option syntax ("[[Let's go here|Destination]]")
+  OptionStart:          /\[\[/, // [[
+  OptionDelimit:        /\|/, // |
+  OptionEnd:            /\]\]/, // ]]
+
+  // Command types (specially recognized command word)
+  If:                   /if(?!\w)/,
+  ElseIf:               /elseif(?!\w)/,
+  Else:                 /else(?!\w)/,
+  EndIf:                /endif(?!\w)/,
+  Jump:                 /jump(?!\w)/,
+  Stop:                 /stop(?!\w)/,
+  Set:                  /set(?!\w)/,
+  Declare:              /declare(?!\w)/,
+  As:                   /as(?!\w)/,
+  ExplicitType:         /(String|Number|Bool)(?=>>)/,
+
+  // Boolean values
+  True:                 /true(?!\w)/,
+  False:                /false(?!\w)/,
+
+  // The null value
+  Null:                 /null(?!\w)/,
+
+  // Parentheses
+  LeftParen:            /\(/,
+  RightParen:           /\)/,
+
+  // Parameter delimiters
+  Comma:                /,/,
+
+  // Operators
+  UnaryMinus:           /-(?!\s)/,
+
+  EqualTo:              /(==|is(?!\w)|eq(?!\w))/, // ==, eq, is
+  GreaterThan:          /(>|gt(?!\w))/, // >, gt
+  GreaterThanOrEqualTo: /(>=|gte(?!\w))/, // >=, gte
+  LessThan:             /(<|lt(?!\w))/, // <, lt
+  LessThanOrEqualTo:    /(<=|lte(?!\w))/, // <=, lte
+  NotEqualTo:           /(!=|neq(?!\w))/, // !=, neq
+
+  // Logical operators
+  Or:                   /(\|\||or(?!\w))/, // ||, or
+  And:                  /(&&|and(?!\w))/, // &&, and
+  Xor:                  /(\^|xor(?!\w))/, // ^, xor
+  Not:                  /(!|not(?!\w))/, // !, not
+
+  // this guy's special because '=' can mean either 'equal to'
+  // or 'becomes' depending on context
+  EqualToOrAssign:      /(=|to(?!\w))/, // =, to
+
+  Add:                  /\+/, // +
+  Minus:                /-/, // -
+  Exponent:             /\*\*/, // **
+  Multiply:             /\*/, // *
+  Divide:               /\//, // /
+  Modulo:               /%/, // /
+
+  AddAssign:            /\+=/, // +=
+  MinusAssign:          /-=/, // -=
+  MultiplyAssign:       /\*=/, // *=
+  DivideAssign:         /\/=/, // /=
+
+  Identifier:           /[a-zA-Z0-9_:.]+/, // a single word (used for functions)
+
+  EscapedCharacter:     /\\./, // for escaping \# special characters
```

Comment:
I found this to be the key to properly handling escapes. In general this file has been modified to only include items in the Yarn 2.0 spec